### PR TITLE
fix bug of outline data process.

### PR DESCRIPTION
**Pull request by bimangle**
When working with outline data, sometimes the 'loadResources.indexBuffersToCreate._array' values is all undefined, cause an exception.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @bimangle!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/master/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @bimangle! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by bimangle** [comment]
**Please verify the merge code, thank you**

QQ图片20210604120003

**Comment by bimangle** [comment]
I can't request a review,  how should I do this?

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @bimangle! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @bimangle! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

File: Source/Scene/ModelOutlineLoader.js [hunk 1]
```diff
@@ -307,6 +307,7 @@ function addOutline(
         // Hackily update it, or else we'll end up creating the wrong type
         // of index buffer later.
         loadResources.indexBuffersToCreate._array.forEach(function (toCreate) {
+          if (toCreate === undefined) return;
```
Comment (ggetz) [inline Source/Scene/ModelOutlineLoader.js:310]: We use the function `Souce/Core/defined` to check if a variable is defined, as it also checks for `null`. So according to the [Coding Guide](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/CodingGuide), this is better written as:

```suggestion
          if (!defined(toCreate)) { 
              return;
          }
```

**Comment by ggetz** [comment]
@kring Would you be able to review this change or recommend a reviewer?

**Comment by kring** [comment]
This PR looks fine to me, certainly no harm. I'd be interested to learn in what circumstances the array has an undefined element, though, because I've never seen that.

**Comment by bimangle** [comment]
thank you for your review.  

We have developed a suite of tools for converting BIM models from multiple source formats (revit, navisworks, microstion ce, sketchup) to 3D Tiles.  

Our tools has an option to include outline when exporting 3D Tiles.  These 3D tiles containing the outline data will randomly cause cesium crashes unless this code is added.  

> This PR looks fine to me, certainly no harm. I'd be interested to learn in what circumstances the array has an undefined element, though, because I've never seen that.

**Comment by kring** [comment]
Thanks @bimangle. If you can share a small glTF or 3D Tiles tileset with us that exhibits the problem, I wouldn't mind taking a look just to make sure this isn't papering over a problem elsewhere. If you don't want to share it publicly, you can share with the email address in my GitHub profile.

**Comment by bimangle** [comment]
> Thanks @bimangle. If you can share a small glTF or 3D Tiles tileset with us that exhibits the problem, I wouldn't mind taking a look just to make sure this isn't papering over a problem elsewhere. If you don't want to share it publicly, you can share with the email address in my GitHub profile.

Thank you for your reply. This is the test data at that time.

Successfully loaded (with cesium v1.78 modified version):
http://fs3.bimangle.net/cesium/t20210304-outline/index-on.html

Failed to load (with cesium v1.89)
http://fs3.bimangle.net/cesium/t20210304-outline/index-error.html

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @bimangle! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by ggetz** [comment]
@bimangle I don't think I have permission to push to your branch here. Would you be able to merge `cesium/main` with your branch?

**Comment by bimangle** [comment]
@ggetz thanks.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @bimangle! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by ggetz** [comment]
Thanks @bimangle.

**Pull request merged**